WordPress/Classes/ViewRelated/Reader/WPImageViewController.m

```diff
@@ -92,37 +122,59 @@ - (void)loadImage
-        __weak __typeof__(self) weakSelf = self;
-        [_imageView setImageWithURLRequest:[NSURLRequest requestWithURL:self.url]
-                         placeholderImage:self.image
-                                  success:^(NSURLRequest *request, NSHTTPURLResponse *response, UIImage *image) {
-                                      __typeof__(self) strongSelf = weakSelf;
-                                      if (!strongSelf) {
-                                          return;
-                                      }
-                                      [strongSelf.activityIndicatorView stopAnimating];
-                                      strongSelf.imageView.image = image;
-                                      [strongSelf.imageView sizeToFit];
-                                      strongSelf.scrollView.contentSize = strongSelf.imageView.image.size;
-                                      [strongSelf centerImage];
-                                      strongSelf.isLoadingImage = NO;
-                                  } failure:^(NSURLRequest *request, NSHTTPURLResponse *response, NSError *error) {
-                                      DDLogError(@"Error loading image: %@", error);
-                                      __typeof__(self) strongSelf = weakSelf;
-                                      [strongSelf.activityIndicatorView stopAnimating];
-                                      strongSelf.isLoadingImage = NO;
-                                  }];
+        [self loadImageFromURL];
+    } else if (self.media) {
+        [self loadImageFromMedia];
     }
 }
 
+- (void)updateImageView
+{
+    self.imageView.image = self.image;
+    [self.imageView sizeToFit];
+    self.scrollView.contentSize = self.imageView.image.size;
+    [self centerImage];
+}
+
+- (void)loadImageFromURL
+{
+    self.isLoadingImage = YES;
+
+    __weak __typeof__(self) weakSelf = self;
+    [_imageView setImageWithURLRequest:[NSURLRequest requestWithURL:self.url]
+                      placeholderImage:self.image
+                               success:^(NSURLRequest *request, NSHTTPURLResponse *response, UIImage *image) {
+                                   weakSelf.image = image;
+                                   [weakSelf updateImageView];
+                                   weakSelf.isLoadingImage = NO;
+                               } failure:^(NSURLRequest *request, NSHTTPURLResponse *response, NSError *error) {
+                                   DDLogError(@"Error loading image: %@", error);
+                                   weakSelf.isLoadingImage = NO;
+                               }];
+}
+
+- (void)loadImageFromMedia
+{
+    self.isLoadingImage = YES;
+
+    __weak __typeof__(self) weakSelf = self;
+    self.imageView.image = self.image;
+    [self.media imageWithSize:CGSizeZero completionHandler:^(UIImage *result, NSError *error) {
+        dispatch_async(dispatch_get_main_queue(), ^{
+            if (error) {
```

Comment:
If you are using weakSelf several times it may be safer to first to convert to strongSelf.